### PR TITLE
Update OpenCensus and Akka (part of [CA-712])

### PR DESCRIPTION
**Pull request by andy7i**
Updates OpenCensus (in order to fix errors, detailed below) and Akka (in order to update OpenCensus, as well as fix issues with running Sam integration tests locally ; includes a small amount of migration)

Related to https://github.com/broadinstitute/sam/pull/421
--------------------

Fixes 500 internal server errors by updating the opencensus versions. while debugging, I found that the code was not handling 204s correctly, leading to the error with "Responses with this status code must have an empty entity". Based on the more recent opencensus code, it looked like this issue was resolved. After updating, most of the several hundred errors were resolved.

Errors looked like:
```
{"causes":[],"exceptionClass":"java.lang.IllegalArgumentException",
"message":"requirement failed: Responses with this status code must have an empty entity","source":"sam","stackTrace":[
{"className":"scala.Predef$","fileName":"Predef.scala","lineNumber":277,"methodName":"require"},
{"className":"akka.http.scaladsl.model.HttpResponse","fileName":"HttpMessage.scala","lineNumber":456,"methodName":"<init>"},
{"className":"akka.http.scaladsl.model.HttpResponse","fileName":"HttpMessage.scala","lineNumber":491,"methodName":"copy"},
{"className":"io.opencensus.scala.akka.http.utils.EndSpanResponse$","fileName":"EndSpanResponse.scala","lineNumber":36,"methodName":"end"},
{"className":"io.opencensus.scala.akka.http.utils.EndSpanResponse$","fileName":"EndSpanResponse.scala","lineNumber":16,"methodName":"end"},
{"className":"io.opencensus.scala.akka.http.TracingDirective","fileName":"TracingDirective.scala","lineNumber":81,"methodName":"$anonfun$recordSuccess$1"},

```

Ticket: https://broadworkbench.atlassian.net/browse/CA-712

---

**PR checklist**

- [x] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [x] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [x] I've updated the [FISMA documentation](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -42,7 +42,6 @@ object Dependencies {
   val ficus: ModuleID =          "com.iheart"                 %% "ficus"           % "1.4.0"
 
   val akkaActor: ModuleID =         "com.typesafe.akka"   %%  "akka-actor"           % akkaV
-  val akkaContrib: ModuleID =       "com.typesafe.akka"   %%  "akka-contrib"         % akkaV
```
Comment (andy7i) [inline project/Dependencies.scala:45]: removed akkaContrib:
the version of akka needed for the latest version of OpenCensus is 2.6.1. Any other version that I put will give an error saying that the versions of akka must be the same. This includes `akkaContrib`, for which the latest version I could find is 2.5.31. Removing it did not affect unit tests and seemingly doesn't affect integration tests either.


Errors with 2.5.31:

```
root [WARN] [04/30/2020 21:33:51.365] [main] [ManifestInfo(akka://sam)] Detected possible incompatible versions on the classpath. Please note that a given Akka version MUST be the same across all modules of Akka that you are using, e.g. if you use [2.6.1] all other modules that are released together MUST be of the same version. Make sure you're using a compatible set of libraries. Possibly conflicting versions [2.5.31, 2.6.1] in libraries [akka-persistence:2.5.31, akka-protobuf:2.5.31, akka-coordination:2.5.31, akka-actor:2.6.1, akka-slf4j:2.6.1, akka-remote:2.5.31, akka-cluster:2.5.31, akka-protobuf-v3:2.6.1, akka-stream:2.6.1, akka-cluster-tools:2.5.31]
root[ERROR] Exception in thread "main" java.lang.IllegalStateException: Detected possible incompatible versions on the classpath. Please note that a given Akka version MUST be the same across all modules of Akka that you are using, e.g. if you use [2.6.1] all other modules that are released together MUST be of the same version. Make sure you're using a compatible set of libraries. Possibly conflicting versions [2.5.31, 2.6.1] in libraries [akka-persistence:2.5.31, akka-protobuf:2.5.31, akka-coordination:2.5.31, akka-actor:2.6.1, akka-slf4j:2.6.1, akka-remote:2.5.31, akka-cluster:2.5.31, akka-protobuf-v3:2.6.1, akka-stream:2.6.1, akka-cluster-tools:2.5.31]
```

**Comment by andy7i** [comment]
jenkins retest -- orch tests aborted

**Comment by andy7i** [comment]
jenkins retest - rawls test failed

**Comment by andy7i** [comment]
jenkins retest - fiab-start failure

File: project/Dependencies.scala [hunk 2]
```diff
@@ -96,10 +95,10 @@ object Dependencies {
...
 
   val excludeScalaCllectionCompat =  ExclusionRule(organization = "org.scala-lang.modules", name = "scala-collection-compat_2.12")
-  val opencensusScalaCode: ModuleID = "com.github.sebruck" %% "opencensus-scala-core" % "0.7.0-M2" excludeAll(excludeScalaCllectionCompat) // excludeAll(excludIoGrpc, excludeCatsEffect )
-  val opencensusAkkaHttp: ModuleID = "com.github.sebruck" %% "opencensus-scala-akka-http" % "0.7.0-M2" excludeAll(excludeScalaCllectionCompat)// excludeAll(excludIoGrpc, excludeCatsEffect)
-  val opencensusStackDriverExporter: ModuleID = "io.opencensus" % "opencensus-exporter-trace-stackdriver" % "0.23.0" // excludeAll(excludIoGrpc, excludeCatsEffect)
-  val opencensusLoggingExporter: ModuleID = "io.opencensus" % "opencensus-exporter-trace-logging"     % "0.23.0" // excludeAll(excludIoGrpc, excludeCatsEffect)
+  val opencensusScalaCode: ModuleID = "com.github.sebruck" %% "opencensus-scala-core" % "0.7.2" // excludeAll(excludIoGrpc, excludeCatsEffect )
+  val opencensusAkkaHttp: ModuleID = "com.github.sebruck" %% "opencensus-scala-akka-http" % "0.7.2" // excludeAll(excludIoGrpc, excludeCatsEffect)
```
Comment (andy7i) [inline project/Dependencies.scala:99]: not sure why `excludeAll(excludeScalaCllectionCompat)` was there but removing it fixes an issue where Stackdriver in opencensus was trying to use collectionconverters but they don't exist in scala 2.12.

File: project/Dependencies.scala [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
-  val akkaHttpV = "10.1.8"
+  val akkaV = "2.6.1"
+  val akkaHttpV = "10.1.11"
   val jacksonV = "2.9.5"
   val scalaLoggingV = "3.5.0"
   val scalaTestV    = "3.0.5"
   val scalaCheckV    = "1.14.0"
   val catsEffectV         = "2.1.1"
-  val scalikejdbcVersion    = "3.3.5"
+  val scalikejdbcVersion    = "3.4.1"
```
Comment (andy7i) [inline project/Dependencies.scala:11]: Fix `NoClassDefFoundError: scala/collection/compat/Factory` issue. https://github.com/scalikejdbc/scalikejdbc/issues/983

  /*
root[ERROR] Exception in thread "pool-2-thread-9" java.lang.NoClassDefFoundError: scala/collection/compat/Factory$
root[ERROR] 	at scalikejdbc.DBSession.list(DBSession.scala:295)
   */

File: automation/project/Dependencies.scala [hunk 4]
```diff
@@ -4,8 +4,8 @@ object Dependencies {
   val scalaV = "2.12"
 
   val jacksonV = "2.8.4"
-  val akkaV = "2.5.7"
-  val akkaHttpV = "10.0.10"
+  val akkaV = "2.6.1"
+  val akkaHttpV = "10.1.11"
```
Comment (andy7i) [inline automation/project/Dependencies.scala:8]: Update automation dependencies, partially to fix an error that comes up when running the tests locally:

```
An exception or error caused a run to abort: spray.json.package$.enrichString(Ljava/lang/String;)Lspray/json/RichString;
java.lang.NoSuchMethodError: spray.json.package$.enrichString(Ljava/lang/String;)Lspray/json/RichString;
	at org.broadinstitute.dsde.workbench.config.CommonConfig$CommonUsers.$init$(CommonConfig.scala:14)
	```

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/Boot.scala [hunk 5]
```diff
@@ -55,7 +54,6 @@ object Boot extends IOApp with LazyLogging {
 
     // we need an ActorSystem to host our application in
     implicit val system = ActorSystem("sam")
-    implicit val materializer = ActorMaterializer()
```
Comment (andy7i) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/Boot.scala:58]: akka 2.6 makes the ActorMaterializer implicit for free with an ActorSystem
https://doc.akka.io/docs/akka/current/project/migration-guide-2.5.x-2.6.x.html#materializer-changes

> Materializer changes
> A default materializer is now provided out of the box. ... for Scala an implicit materializer is provided if there is an implicit ActorSystem available. This avoids leaking materializers and simplifies most stream use cases somewhat.
> 
> The ActorMaterializer factories has been deprecated and replaced with a few corresponding factories in akka.stream.Materializer. New factories with per-materializer settings has not been provided but should instead be done globally through config or per stream, see below for more details.
> 
> Having a default materializer available means that most, if not all, usages of Java ActorMaterializer.create() and Scala implicit val materializer = ActorMaterializer() should be removed.

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/service/StatusService.scala [hunk 6]
```diff
@@ -26,7 +26,7 @@ class StatusService(
   implicit val askTimeout = Timeout(5 seconds)
 
   private val healthMonitor = system.actorOf(HealthMonitor.props(cloudExtensions.allSubSystems + OpenDJ)(checkStatus _))
-  system.scheduler.schedule(initialDelay, pollInterval, healthMonitor, HealthMonitor.CheckAll)
```
Comment (andy7i) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/service/StatusService.scala:29]: schedule is deprecated and scheduleAtFixedRate is the recommended, performant replacement

**Review by dvoet — APPROVED** [review]

**Review by gpolumbo-broad — APPROVED** [review]

**Pull request merged**